stripe/src/main/java/com/stripe/android/view/CardMultilineWidget.java

```diff
@@ -0,0 +1,399 @@
+import android.text.InputFilter;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.View;
+import android.widget.LinearLayout;
+
+import com.stripe.android.R;
+import com.stripe.android.model.Card;
+import com.stripe.android.CardUtils;
+
+import static com.stripe.android.model.Card.BRAND_RESOURCE_MAP;
+import static com.stripe.android.view.CardInputListener.FocusField.FOCUS_CARD;
+import static com.stripe.android.view.CardInputListener.FocusField.FOCUS_CVC;
+import static com.stripe.android.view.CardInputListener.FocusField.FOCUS_EXPIRY;
+import static com.stripe.android.view.CardInputListener.FocusField.FOCUS_POSTAL;
+
+/**
+ * A multiline card input widget using the support design library's {@link TextInputLayout}
+ * to match Material Design.
+ */
+public class CardMultilineWidget extends LinearLayout {
+
+    static final String CARD_MULTILINE_TOKEN = "CardMultilineView";
+
+    private @Nullable CardInputListener mCardInputListener;
+    private CardNumberEditText mCardNumberEditText;
+    private ExpiryDateEditText mExpiryDateEditText;
+    private StripeEditText mCvcEditText;
+    private StripeEditText mPostalCodeEditText;
+    private TextInputLayout mCvcTextInputLayout;
+
+    private boolean mShouldShowPostalCode;
+    private @Card.CardBrand String mCardBrand;
+    private @ColorInt int mTintColorInt;
+
+    public CardMultilineWidget(Context context) {
+        super(context);
+        initView(null);
+    }
+
+    public CardMultilineWidget(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initView(attrs);
+    }
+
+    public CardMultilineWidget(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initView(attrs);
+    }
+
+    @VisibleForTesting
+    CardMultilineWidget(Context context, boolean shouldShowPostalCode) {
+        super(context);
+        mShouldShowPostalCode = shouldShowPostalCode;
+        initView(null);
+    }
+
+    /**
+     * @param cardInputListener A {@link CardInputListener} to be notified of changes
+     *                          to the user's focused field
+     */
+    public void setCardInputListener(@Nullable CardInputListener cardInputListener) {
+        mCardInputListener = cardInputListener;
+    }
+
+    /**
+     * Gets a {@link Card} object from the user input, if all fields are valid. If not, returns
+     * {@code null}.
+     *
+     * @return a valid {@link Card} object based on user input, or {@code null} if any field is
+     * invalid
+     */
+    @Nullable
+    public Card getCard() {
+        String cardNumber = mCardNumberEditText.getCardNumber();
+        int[] cardDate = mExpiryDateEditText.getValidDateFields();
+        if (cardNumber == null || cardDate == null || cardDate.length != 2) {
+            return null;
+        }
+
+        String cvcValue = mCvcEditText.getText().toString();
+        if (validateAllFields()) {
```

Comment:
Do you want to lock in on an preference for early returns as a style preference? If so, perhaps we can covert this to an early return?